File: testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/report_model/ReportHooks.kt
package com.testerum.runner.report_model

import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.feature.hooks.HookPhase
import com.testerum_api.testerum_steps_api.test_context.ExecutionStatus
import java.time.LocalDateTime

data class ReportHooks(@JsonProperty("hookPhase")        val hookPhase: HookPhase,
                       @JsonProperty("startTime")        val startTime: LocalDateTime,
                       @JsonProperty("endTime")          val endTime: LocalDateTime,
                       @JsonProperty("durationMillis")   val durationMillis: Long,
                       @JsonProperty("status")           val status: ExecutionStatus,
                       @JsonProperty("textLogFilePath")  override val textLogFilePath: String,
                       @JsonProperty("modelLogFilePath") override val modelLogFilePath: String,
                       @JsonProperty("children")         val children: List<ReportStep>) : FeatureOrTestRunnerReportNode

File: testerum-backend/testerum-runner/testerum-runner-cmdline/src/main/assembly/report_templates/common/testerum-model/report-model/model/report/report-hooks.ts
import {FeatureOrTestRunnerReportNode} from "./feature-or-test-runner-report-node";
import {ExecutionStatus} from "./execution-status";
import {MarshallingUtils} from "../../../json-marshalling/marshalling-utils";
import {HookPhase} from "./hook-phase";
import {ReportStep} from "./report-step";

export class ReportHooks implements FeatureOrTestRunnerReportNode {

    constructor(public readonly hookPhase: HookPhase,
                public readonly startTime: Date,
                public readonly endTime: Date,
                public readonly durationMillis: number,
                public readonly status: ExecutionStatus,
                public readonly textLogFilePath: string,
                public readonly modelLogFilePath: string,
                public readonly children: Array<ReportStep>) {
    }

    static parse(input: Object): ReportHooks {
        if (!input) {
            return null;
        }

        const hookPhase = MarshallingUtils.parseEnum(input["hookPhase"], HookPhase);
        const startTime = MarshallingUtils.parseUtcToLocalDateTime(input["startTime"]);
        const endTime = MarshallingUtils.parseUtcToLocalDateTime(input["endTime"]);
        const durationMillis = input["durationMillis"];
        const status = MarshallingUtils.parseEnum(input["status"], ExecutionStatus);
        const textLogFilePath = input["textLogFilePath"];
        const modelLogFilePath = input["modelLogFilePath"];
        const children = MarshallingUtils.parseList(input["children"], ReportStep);

        return new ReportHooks(hookPhase, startTime, endTime, durationMillis, status, textLogFilePath, modelLogFilePath, children);
    }

}
